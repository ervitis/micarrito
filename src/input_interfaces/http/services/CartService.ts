
export class CartService {
  private cartRepository

  constructor ({ cartRepository }) {
    this.cartRepository = cartRepository
  }

  async execute () {
    try {
      const data = await this.cartRepository.repositoryGetAllItems()
      return data
    } catch (err) {
      throw err
    }

  }
}
