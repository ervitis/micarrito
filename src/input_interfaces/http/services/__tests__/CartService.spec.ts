import { CartService } from '../CartService'

describe('CartService', () => {
  const cartRepository = {
    repositoryGetAllItems: jest.fn(() => Promise.resolve([{id: 1, name: 'hello'}]))
  }

  test('Instance', () => {
    const cartService = new CartService({ cartRepository })
    expect(cartService).toBeDefined()
  })

  test('Get all items', async () => {
    const cartService = new CartService({ cartRepository })
    const data = await cartService.execute()

    expect(data).toBeDefined()
    expect(data).toMatchSnapshot()
    expect(data.length).toBe(1)
  })
})
