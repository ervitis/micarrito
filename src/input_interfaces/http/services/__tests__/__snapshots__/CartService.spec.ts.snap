// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartService Get all items 1`] = `
Array [
  Object {
    "id": 1,
    "name": "hello",
  },
]
`;
